packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -298,13 +299,17 @@ interface BundlingCommandOptions {
 class OsCommand {
   constructor(private readonly osPlatform: NodeJS.Platform) {}
 
-  public writeJson(filePath: string, data: any): string {
-    const stringifiedData = JSON.stringify(data);
+  public write(filePath: string, data: any): string {
     if (this.osPlatform === 'win32') {
-      return `echo ^${stringifiedData}^ > "${filePath}"`;
+      return `echo ^${data}^ > "${filePath}"`;
     }
 
-    return `echo '${stringifiedData}' > "${filePath}"`;
+    return `echo '${data}' > "${filePath}"`;
+  }
+
+  public writeJson(filePath: string, data: any): string {
```

Comment:
And why this doesn't need the extra check for OS?

Reply:
The OS check has been moved to the `write()` function